sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/lang/IntermediateType.kt

```diff
@@ -140,15 +140,24 @@ internal data class IntermediateType(
     return cursorGetter
   }
 
-  enum class SqliteType(val javaType: TypeName) {
+  interface DialectType {
```

Comment:
verrrrry nice.

wanna shove this into a new package somewhere like `dialect.api` so we can start to consolidate things that will eventually need to be moved to sql-psi?

Reply:
Sure thing! I suppose putting it in `com.squareup.sqldelight.core.dialect.api` makes the most sense for now. Should we also move the dialect specific types to another package, e.g., `PostgreSqlType` to `*.dialect.postgresql`?

Reply:
yep that makes sense to me

Reply:
Fixed in https://github.com/cashapp/sqldelight/pull/2014/commits/aa6bf3213835ad854e3e92d4b69b3a1da7501940